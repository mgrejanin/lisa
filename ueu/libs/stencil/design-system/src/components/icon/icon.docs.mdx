import { Story, Icons, Status } from './../../../.storybook/components';
import { Props } from '@storybook/addon-docs/blocks';

# Icons

<Status />

Os ícones são usados ​​no produto para apresentar ações e comandos comuns.

<Story id="design-tokens-icons--base" />

<Props id="design-tokens-icons--base" />

Ícones neste pacote suporta os seguintes tamanhos: `16` e `24` pixels. Esses tamanhos referem-se à largura e altura do ícone. Você pode importar um componente de ícone para o seu projeto, referindo-se ao seu nome e tamanho:

```html
<apollo-icon svgIcon="arrows-arrow-up" size="sm"></apollo-icon>
```

### Acessibilidade

Por padrão, os componentes de ícone são tratados como conteúdo decorativo. Se desejar que o ícone seja anunciado por um leitor de tela, você pode fornecer um `label`. Exemplo:

```html
<apollo-icon svgIcon="interface-plus-circle" size="sm" label="Adicionar"></apollo-icon>
```

Se você precisar adicionar uma descrição de funcionalidade para o ícone você pode fornecer um `role`. Exemplo:

```html
<apollo-icon svgIcon="file-file-blank" size="sm" role="document"></apollo-icon>
```

### Material Icons

Evite o uso de material icons. Sempre confira se existe no Design System ícone que atende a necessidade ou a possibilidade de inclusão de novos ícones no DS.
Caso seja necessário usar ícone do material você pode usar conforme exemplo abaixo:

<Story id="design-tokens-icons--slot" />

### Lista completa de ícones

<Icons />
