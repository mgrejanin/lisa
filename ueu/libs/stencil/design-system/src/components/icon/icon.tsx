import { Component, h, Host, Prop, State } from '@stencil/core';
import { setPropStyle } from '../../helpers/set-prop-style';

@Component({
    tag: 'apollo-icon',
    styleUrl: 'icon.scss',
    shadow: false,
})
export class Icon {
    /**
     * O tamanho do ícone
     */
    @Prop() size: 'sm' | 'md' | 'lg' = 'md';

    /**
     * Cor do ícone. Tokens DS.
     */
    @Prop() color: string;

    /**
     * O nome do ícone SVG.
     */
    @Prop()
    svgIcon: string;

    /**
     * O tipo do ícone SVG
     */
    @Prop({ mutable: true })
    typeIcon: string;

    /**
     * O atributo que define a função.
     */
    @Prop()
    role: string;

    /**
     * Texto de ajuda. Para acessibilidade.
     */
    @Prop()
    label: string;

    /**
     * Cor do ícone. Hexadecimal.
     */
    @Prop()
    fill: string;

    /**
     * O atributo diâmetro (definirá a largura e a altura do SVG).
     */
    @State() diameter: number;

    componentWillRender() {
        if (this.svgIcon) {
            let icon = this.svgIcon;
            let type = icon.split('-');

            this.typeIcon = type[0];
        }

        this.renderSize();
    }

    private renderSize() {
        switch (this.size) {
            case 'sm':
                this.diameter = 16;
                break;
            case 'md':
                this.diameter = 24;
                break;
            case 'lg':
                this.diameter = 40;
                break;
            default:
                this.diameter = 16;
                break;
        }
    }

    render() {
        if (this.svgIcon === undefined && this.typeIcon === undefined) {
            return (
                <Host style={{ fontSize: this.size, color: this.fill }} class={setPropStyle(this)} aria-hidden="true">
                    <slot></slot>
                </Host>
            );
        }

        return (
            <Host class="apollo-icon--svg">
                <svg
                    viewBox={`0 0 ${this.diameter} ${this.diameter}`}
                    width={this.diameter}
                    height={this.diameter}
                    role={this.role}
                    fill={this.fill}
                    color={this.fill}
                    aria-label={this.label}
                    class={setPropStyle(this)}
                >
                    <use xlinkHref={`./assets/design-system/icons/${this.typeIcon}.svg#${this.svgIcon}`} />
                </svg>
            </Host>
        );
    }
}
