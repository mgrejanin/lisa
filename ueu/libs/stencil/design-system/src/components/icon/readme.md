# apollo-icon



<!-- Auto Generated Below -->


## Properties

| Property   | Attribute   | Description                          | Type                   | Default     |
| ---------- | ----------- | ------------------------------------ | ---------------------- | ----------- |
| `color`    | `color`     | Cor do ícone. Tokens DS.             | `string`               | `undefined` |
| `fill`     | `fill`      | Cor do ícone. Hexadecimal.           | `string`               | `undefined` |
| `label`    | `label`     | Texto de ajuda. Para acessibilidade. | `string`               | `undefined` |
| `role`     | `role`      | O atributo que define a função.      | `string`               | `undefined` |
| `size`     | `size`      | O tamanho do ícone                   | `"lg" \| "md" \| "sm"` | `'md'`      |
| `svgIcon`  | `svg-icon`  | O nome do ícone SVG.                 | `string`               | `undefined` |
| `typeIcon` | `type-icon` | O tipo do ícone SVG                  | `string`               | `undefined` |


## Dependencies

### Used by

 - [apollo-button](../button)
 - [apollo-dialog](../dialog)
 - [apollo-feedback-card](../feedback-card)
 - [apollo-feedback-page](../feedback-page)
 - [apollo-icon-button](../icon-button)
 - [apollo-modal](../modal)
 - [apollo-search-bar](../surfaces/toolbar/search-bar)
 - [apollo-sidebar](../navigation/sidebar)
 - [apollo-snackbar](../snackbar)
 - [apollo-textfield](../textfield)
 - [apollo-toggle-button](../toggle-button)
 - [textfield-helper-text](../textfield/helper-text)

### Graph
```mermaid
graph TD;
  apollo-button --> apollo-icon
  apollo-dialog --> apollo-icon
  apollo-feedback-card --> apollo-icon
  apollo-feedback-page --> apollo-icon
  apollo-icon-button --> apollo-icon
  apollo-modal --> apollo-icon
  apollo-search-bar --> apollo-icon
  apollo-sidebar --> apollo-icon
  apollo-snackbar --> apollo-icon
  apollo-textfield --> apollo-icon
  apollo-toggle-button --> apollo-icon
  textfield-helper-text --> apollo-icon
  style apollo-icon fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

PicPay Doc
