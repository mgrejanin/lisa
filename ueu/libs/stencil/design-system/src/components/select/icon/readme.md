# select-icon

<!-- Auto Generated Below -->


## Properties

| Property    | Attribute    | Description | Type      | Default |
| ----------- | ------------ | ----------- | --------- | ------- |
| `ariaLabel` | `aria-label` | aria-label  | `string`  | `''`    |
| `disabled`  | `disabled`   | disabled    | `boolean` | `false` |


----------------------------------------------

PicPay Doc
