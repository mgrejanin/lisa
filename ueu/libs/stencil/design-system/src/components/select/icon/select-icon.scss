@use "@material/theme" with (
    $primary: theme('colors.primary.light.500'),
    $secondary: theme('colors.primary.light.500'),
);

@use "@material/select/icon";

@include icon.icon-core-styles;

select-icon {
    display: flex;
    align-items: center;
    align-content: center;
    justify-content: center;
    outline: none;
}

.mdc-select {
    .mdc-select__icon {
        line-height: 1;
        height: 24px;
        margin-left: 16px;
        margin-right: 8px;
    }
    .mdc-list-item {
        .mdc-select__icon {
            margin-left: 0px;
            margin-right: 8px;
        }
    }
    &.mdc-select--disabled {
        @include icon.disabled-icon-color(rgba(0, 0, 0, theme.text-emphasis(disabled)));
    }
    &:not(.mdc-select--invalid):not(.mdc-select--disabled) {
        @include icon.icon-color(primary);
        .mdc-list-item {
            @include icon.icon-color(rgba(0, 0, 0, theme.text-emphasis(disabled)));
            &.mdc-list-item--selected {
                @include icon.icon-color(primary);
            }
        }
    }
}

.mdc-select--invalid {
    @include icon.icon-color(theme('colors.critical.base'));
    .mdc-list-item {
        @include icon.icon-color(rgba(0, 0, 0, theme.text-emphasis(disabled)));
        &.mdc-list-item--selected {
            @include icon.icon-color(primary);
        }
    }
}

[data-theme='dark'] {
    .mdc-select {
        &.mdc-select--disabled {
            @include icon.disabled-icon-color(rgba(0, 0, 0, theme.text-emphasis(disabled)));
        }

        &:not(.mdc-select--invalid):not(.mdc-select--disabled) {
            @include icon.icon-color(primary);
            .mdc-list-item {
                @include icon.icon-color(theme('colors.grayscale.400'));
                &.mdc-list-item--selected {
                    @include icon.icon-color(theme('colors.grayscale.400'));
                }
            }
        }
    }

    .mdc-select--invalid {
        @include icon.icon-color(theme('colors.critical.base'));
        .mdc-list-item {
            @include icon.icon-color(theme('colors.grayscale.400'));
            &.mdc-list-item--selected {
                @include icon.icon-color(theme('colors.grayscale.400'));
            }
        }
    }
}
