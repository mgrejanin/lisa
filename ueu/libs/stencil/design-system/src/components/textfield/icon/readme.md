# textfield-icon

<!-- Auto Generated Below -->


## Properties

| Property    | Attribute    | Description | Type                      | Default     |
| ----------- | ------------ | ----------- | ------------------------- | ----------- |
| `ariaLabel` | `aria-label` | aria-label  | `string`                  | `''`        |
| `disabled`  | `disabled`   | disabled    | `boolean`                 | `false`     |
| `type`      | `type`       | type        | `"leading" \| "trailing"` | `'leading'` |


## Events

| Event         | Description     | Type                         |
| ------------- | --------------- | ---------------------------- |
| `apolloKeyUp` | Evento de keyUp | `CustomEvent<KeyboardEvent>` |


## Dependencies

### Used by

 - [apollo-search-bar](../../surfaces/toolbar/search-bar)
 - [apollo-textfield](..)

### Graph
```mermaid
graph TD;
  apollo-search-bar --> apollo-textfield-icon
  apollo-textfield --> apollo-textfield-icon
  style apollo-textfield-icon fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

PicPay Doc
