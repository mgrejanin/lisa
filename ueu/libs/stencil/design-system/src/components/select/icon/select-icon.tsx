import { MDCSelectIcon } from '@material/select/icon';
import { Component, h, Host, Prop, Watch } from '@stencil/core';
import classnames from 'classnames';

@Component({
    tag: 'apollo-select-icon',
    styleUrl: 'select-icon.scss',
    shadow: false,
})
export class ApolloSelectIcon {
    /**
     * disabled
     */
    @Prop() disabled = false;

    /**
     * aria-label
     */
    @Prop() ariaLabel = '';
    private iconEl: HTMLElement;
    private icon: MDCSelectIcon;
    @Watch('disabled')
    setDisabled(isDisabled: boolean) {
        this.icon.getDefaultFoundation().setDisabled(isDisabled);
    }
    @Watch('ariaLabel')
    setAriaLabel(aria: string) {
        this.icon.getDefaultFoundation().setAriaLabel(aria);
    }

    componentDidLoad() {
        this.icon = new MDCSelectIcon(this.iconEl);
    }

    render() {
        return (
            <Host
                class={classnames('mdc-select__icon', 'mdc-select__icon--leading')}
                tabindex="0"
                role="button"
                ref={(el: HTMLElement) => (this.iconEl = el)}
            >
                <slot></slot>
            </Host>
        );
    }
}
