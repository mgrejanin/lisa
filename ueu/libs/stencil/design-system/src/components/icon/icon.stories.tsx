import { h } from '@stencil/core';

import readme from './readme.md';
import docs from './icon.docs.mdx';

const sizeOptions = ['sm', 'md'];

export default {
    title: 'Design Tokens/Icons',
    component: 'apollo-icon',
    parameters: {
        docs: { page: docs },
        notes: readme,
    },
    argTypes: {
        color: {
            description: 'Cor do ícone. Deve ser utilizado com tokens do DS.',
            control: { type: 'text' },
        },
        fill: {
            description: 'Cor do ícone. Deve ser utilizado quando for preciso usar hexa.',
            control: { type: 'color' },
        },
        label: {
            description: 'Texto de ajuda. Para acessibilidade.',
            control: { type: 'text' },
        },
        role: {
            description: 'Descrição de funcionalidade.',
            control: { type: 'text' },
        },
        size: {
            description: 'O tamanho do ícone',
            table: {
                defaultValue: { summary: 'md' },
            },
            control: {
                type: 'select',
                options: sizeOptions,
            },
        },
        'svg-icon': {
            description: 'O nome do ícone SVG.',
            control: { type: 'text' },
        },
    },
};

export const Base = (args: any) => {
    return <apollo-icon {...args}></apollo-icon>;
};

Base.args = {
    'svg-icon': 'interface-bell',
    size: 'md',
};

export const Slot = (args: any) => {
    return (
        <apollo-icon {...args} class="material-icon">
            keyboard_arrow_right
        </apollo-icon>
    );
};

Slot.args = {
    color: 'brand.base',
    size: 'md',
};
