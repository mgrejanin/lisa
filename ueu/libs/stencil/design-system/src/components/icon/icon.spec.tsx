import { newSpecPage, SpecPage } from '@stencil/core/testing';
import { Icon } from './icon';

describe('IconComponent', () => {
    let page: SpecPage;
    let element: HTMLApolloIconElement;

    beforeEach(async () => {
        page = await newSpecPage({
            components: [Icon],
            html: `<apollo-icon svg-icon="arrows-arrow-circle-left"></apollo-icon>`,
        });

        element = page.doc.querySelector('apollo-icon');

        expect(page.root).toMatchSnapshot();
    });

    it('should build', () => {
        expect(element).toBeTruthy();
        expect(element).toHaveClass('apollo-icon--svg');
    });

    it('should render props', () => {
        expect(element.svgIcon).toEqual('arrows-arrow-circle-left');
        expect(element.size).toEqual('md');
    });

    it('should have expected viewBox on <svg>', () => {
        expect(element.querySelector('svg').attributes.getNamedItem('viewBox')).not.toEqual('');
        expect(element.querySelector('svg').attributes.getNamedItem('viewBox').value).toEqual('0 0 24 24');
    });

    it('should have expected width, height on <svg>', () => {
        expect(element.querySelector('svg').attributes.getNamedItem('width').value).toEqual('24');
        expect(element.querySelector('svg').attributes.getNamedItem('height').value).toEqual('24');
    });

    it('should have expected color and fill on <svg>', async () => {
        element.fill = '#ccc';

        await page.waitForChanges();

        expect(element.querySelector('svg').attributes.getNamedItem('color').value).toEqual('#ccc');
        expect(element.querySelector('svg').attributes.getNamedItem('fill').value).toEqual('#ccc');
    });

    it('should have expected role on <svg>', async () => {
        element.role = 'document';

        await page.waitForChanges();

        expect(element.querySelector('svg').attributes.getNamedItem('role').value).toEqual('document');
    });

    it('should have expected label on <svg>', async () => {
        element.label = 'Documentar';

        await page.waitForChanges();

        expect(element.querySelector('svg').attributes.getNamedItem('aria-label').value).toEqual('Documentar');
    });

    it('should be able to change the type and svg', async () => {
        element.svgIcon = 'brand-google';
        element.size = 'sm';

        await page.waitForChanges();

        expect(element.svgIcon).toEqual('brand-google');
        expect(element.size).toEqual('sm');
        expect(element.querySelector('svg').attributes.getNamedItem('width').value).toEqual('16');
        expect(element.querySelector('svg').attributes.getNamedItem('height').value).toEqual('16');
        expect(element.querySelector('svg').attributes.getNamedItem('viewBox')).not.toEqual('');
        expect(element.querySelector('svg').attributes.getNamedItem('viewBox').value).toEqual('0 0 16 16');
    });

    it('must be able to apply the default size', async () => {
        element.size = null;

        await page.waitForChanges();

        expect(element.querySelector('svg').attributes.getNamedItem('width').value).toEqual('16');
        expect(element.querySelector('svg').attributes.getNamedItem('height').value).toEqual('16');
        expect(element.querySelector('svg').attributes.getNamedItem('viewBox')).not.toEqual('');
        expect(element.querySelector('svg').attributes.getNamedItem('viewBox').value).toEqual('0 0 16 16');
    });
});
