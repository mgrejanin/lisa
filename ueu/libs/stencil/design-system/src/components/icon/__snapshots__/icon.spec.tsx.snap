// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconComponent must be able to apply the default size 1`] = `
<apollo-icon class="apollo-icon--svg" svg-icon="arrows-arrow-circle-left">
  <!---->
  <svg class="css-1h3erg3" height="24" viewBox="0 0 24 24" width="24">
    <use xlink:href="./assets/design-system/icons/arrows.svg#arrows-arrow-circle-left"></use>
  </svg>
</apollo-icon>
`;

exports[`IconComponent should be able to change the type and svg 1`] = `
<apollo-icon class="apollo-icon--svg" svg-icon="arrows-arrow-circle-left">
  <!---->
  <svg class="css-1h3erg3" height="24" viewBox="0 0 24 24" width="24">
    <use xlink:href="./assets/design-system/icons/arrows.svg#arrows-arrow-circle-left"></use>
  </svg>
</apollo-icon>
`;

exports[`IconComponent should build 1`] = `
<apollo-icon class="apollo-icon--svg" svg-icon="arrows-arrow-circle-left">
  <!---->
  <svg class="css-1h3erg3" height="24" viewBox="0 0 24 24" width="24">
    <use xlink:href="./assets/design-system/icons/arrows.svg#arrows-arrow-circle-left"></use>
  </svg>
</apollo-icon>
`;

exports[`IconComponent should have expected color and fill on <svg> 1`] = `
<apollo-icon class="apollo-icon--svg" svg-icon="arrows-arrow-circle-left">
  <!---->
  <svg class="css-1h3erg3" height="24" viewBox="0 0 24 24" width="24">
    <use xlink:href="./assets/design-system/icons/arrows.svg#arrows-arrow-circle-left"></use>
  </svg>
</apollo-icon>
`;

exports[`IconComponent should have expected label on <svg> 1`] = `
<apollo-icon class="apollo-icon--svg" svg-icon="arrows-arrow-circle-left">
  <!---->
  <svg class="css-1h3erg3" height="24" viewBox="0 0 24 24" width="24">
    <use xlink:href="./assets/design-system/icons/arrows.svg#arrows-arrow-circle-left"></use>
  </svg>
</apollo-icon>
`;

exports[`IconComponent should have expected role on <svg> 1`] = `
<apollo-icon class="apollo-icon--svg" svg-icon="arrows-arrow-circle-left">
  <!---->
  <svg class="css-1h3erg3" height="24" viewBox="0 0 24 24" width="24">
    <use xlink:href="./assets/design-system/icons/arrows.svg#arrows-arrow-circle-left"></use>
  </svg>
</apollo-icon>
`;

exports[`IconComponent should have expected viewBox on <svg> 1`] = `
<apollo-icon class="apollo-icon--svg" svg-icon="arrows-arrow-circle-left">
  <!---->
  <svg class="css-1h3erg3" height="24" viewBox="0 0 24 24" width="24">
    <use xlink:href="./assets/design-system/icons/arrows.svg#arrows-arrow-circle-left"></use>
  </svg>
</apollo-icon>
`;

exports[`IconComponent should have expected width, height on <svg> 1`] = `
<apollo-icon class="apollo-icon--svg" svg-icon="arrows-arrow-circle-left">
  <!---->
  <svg class="css-1h3erg3" height="24" viewBox="0 0 24 24" width="24">
    <use xlink:href="./assets/design-system/icons/arrows.svg#arrows-arrow-circle-left"></use>
  </svg>
</apollo-icon>
`;

exports[`IconComponent should render props 1`] = `
<apollo-icon class="apollo-icon--svg" svg-icon="arrows-arrow-circle-left">
  <!---->
  <svg class="css-1h3erg3" height="24" viewBox="0 0 24 24" width="24">
    <use xlink:href="./assets/design-system/icons/arrows.svg#arrows-arrow-circle-left"></use>
  </svg>
</apollo-icon>
`;
